File: backend/src/main/java/org/akashihi/mdg/api/v1/TagController.java
package org.akashihi.mdg.api.v1;

import lombok.RequiredArgsConstructor;
import org.akashihi.mdg.api.v1.dto.Tags;
import org.akashihi.mdg.service.TransactionService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class TagController {
    private final TransactionService transactionService;

    @GetMapping(value = "/tags", produces = "application/vnd.mdg+json;version=1")
    public Tags list() {
        return new Tags(transactionService.listTags());
    }
}

File: backend/src/main/java/org/akashihi/mdg/api/v1/TagController.kt
package org.akashihi.mdg.api.v1

import org.akashihi.mdg.entity.Tag
import org.akashihi.mdg.service.TransactionService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

data class Tags(val tags: Collection<Tag>)

@RestController
class TagController(private val transactionService: TransactionService) {
    @GetMapping(value = ["/tags"], produces = ["application/vnd.mdg+json;version=1"])
    fun list(): Tags = Tags(transactionService.listTags())
}
